conda_smithy/configure_feedstock.py

```diff
@@ -1328,6 +1331,19 @@ def render_README(jinja_env, forge_config, forge_dir, render_info=None):
             forge_config["azure"]["build_id"] = build_def["id"]
         except (IndexError, IOError):
             pass
+        except json.decoder.JSONDecodeError:
+            try:
+                # If it fails then we switch to a request using an Azure token.
+                config = azure_ci_utils.AzureConfig(
+                    org_or_user=forge_config["azure"]["user_or_org"],
+                    project_name=forge_config["azure"]["project_name"],
+                )
+                repo = forge_config["github"]["repo_name"]
+                build_info = azure_ci_utils.get_build_id(repo, config)
+                forge_config["azure"]["build_id"] = build_info["build_id"]
+            except:
```

Comment:
I just think that this big try catch which is getting the whole exceptions might shallow some problems.
And I think it would be nice to add the exception message to the log when you are catching all the exceptions

Reply:
Please check again and let me know.